pkg/kademlia/endpoint.go

```diff
@@ -35,8 +36,14 @@ func NewEndpoint(log *zap.Logger, service *Kademlia, routingTable *RoutingTable)
 
 // Query is a node to node communication query
 func (endpoint *Endpoint) Query(ctx context.Context, req *pb.QueryRequest) (*pb.QueryResponse, error) {
+
+	duration := time.Duration(req.QueryTimeout.Seconds) * time.Second
```

Comment:
`QueryTimeout` should be specified for `Endpoint` not the request.

Reply:
oh okay